### ISSUE TITLE
Too many thread pools

### ISSUE THREAD

**Issue by mattes** [ursm/activejob-google_cloud_pubsub#4]
[google-cloud-pubsub](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/google-cloud-pubsub/v0.30.0/google-cloud-pubsub/lib/google/cloud/pubsub/subscriber.rb#L108-L110) is starting a thread pool to process incoming messages:

```ruby
            @stream_pool.map do |stream|
              Thread.new { stream.start }
            end
```

[activejob-google_cloud_pubsub](https://github.com/ursm/activejob-google_cloud_pubsub/blob/master/lib/activejob-google_cloud_pubsub/worker.rb#L25) is doing the same:

```ruby
pool = Concurrent::ThreadPoolExecutor.new(min_threads: @min_threads, max_threads: @max_threads, max_queue: -1)
```

... effectively duplicating the efforts of google-cloud-pubsub, if I'm not mistaken?

**Comment by ursm** [comment]
Exactly. Earlier versions of google-cloud-pubsub did not use thread pools, but now it's an extra feature. I am going to fix this. Thank you very much for your attention!

**Comment by hellupline** [comment]
could #6 be the solution for this issue ?
